### ISSUE TITLE
Error when trying to use the example Hive tokenstore

### ISSUE THREAD

**Issue by MostHated** [cachapa/firedart#34]
Edit ------- It looks like, for the last part of this post, within the Token class, changing it to the following fixed it: There are still the few other things below, which are just small tweaks. I am not sure if having to do HiveStore.create() as opposed to just using HiveStore() is intended. It's getting a bit late for me tonight, but I can try to do a PR tomorrow unless you get to it first.
 
```dart
Token.fromMap(Map<dynamic, dynamic> map)

// and also
Map<dynamic, dynamic> toMap()
```

---
### // Original 
Hey there,  
 I came across this package today and was rather excited to see that I won't have to use the web API Frankenstein setup I have been using, lol. I wanted to give Hive a try as well, as I had not heard of it prior(I was currently just using the package cross_local_storage as it works on Desktop), it looks like there may have been a change or two since the example was made? 

When I first put it in my project, I got warning and the following was generated for the TokenAdapter extends TypeAdapter<Token> class:

```dart
  @override
  // TODO: implement typeId
  int get typeId => throw UnimplementedError();
```

It also looks like Hive.registerAdapter() may now just be:
```dart
void registerAdapter<T>(TypeAdapter<T> adapter)
```

The example also showed:
```dart
FirebaseAuth.initialize(apiKey, await HiveStore());
```
But it was saying there was no constructor, but it accepted:
```dart
FirebaseAuth.initialize(apiKey, await HiveStore.create());
```
All errors were gone at that point and I went ahead to run it. Everything fired up just fine, my test account logged in, an auth_store.hive/.lock file were generated, which I am assuming meant all was good, as there were no errors or anything.

I closed and reran my app, at which point I was greeted by the following:
```terminal
flutter: ┌───────────────────────────────────────────────────────────────────────────────────────────────────
flutter: │ type '_InternalLinkedHashMap<dynamic, dynamic>' is not a subtype of type 'Map<String, dynamic>'
flutter: ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
flutter: │ #0   package:searcher_installer_go/data/models/hive_store.dart 41:59  TokenAdapter.read
flutter: │ #1   package:hive/src/binary/binary_reader_impl.dart 325:33           BinaryReaderImpl.read
flutter: │ #2   package:hive/src/binary/binary_reader_impl.dart 273:26           BinaryReaderImpl.readFrame
flutter: │ #3   package:hive/src/binary/frame_helper.dart 17:26                  FrameHelper.framesFromBytes
flutter: │ #4   package:hive/src/io/frame_io_helper.dart 41:12                   FrameIoHelper.framesFromFile
flutter: │ #5   ===== asynchronous gap ===========================
flutter: │ #6   dart:async                                                       _asyncErrorWrapperHelper
flutter: │ #7   package:searcher_installer_go/main.dart 106:79                   runMain.<fn>
flutter: │ #8   package:provider/src/inherited_provider.dart 576:35              _CreateInheritedProviderState.value
flutter: │ #9   package:provider/src/inherited_provider.dart 439:33              _InheritedProviderScopeElement.value
flutter: ├┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
```

Thanks!
-MH

**Comment by cachapa** [comment]
Thanks for the report.
I didn't want to add a dependency to Hive to the project, that's why it's linked as a gist. Unfortunately that means I have to remember to keep the gist up to date with the changes in the API.

> It also looks like Hive.registerAdapter() may now just be:

Yeah the Hive API changed in the mean time.

> FirebaseAuth.initialize(apiKey, await HiveStore());

That's my bad. The reason for `await HiveStore.create()` is that dart doesn't allow for async constructors.

> I closed and reran my app, at which point I was greeted by the following:

That's a Hive error due to Dart's type erasure. For some reason it was working when I made that gist, but something changed in the meanwhile.

I'll update the example, thanks for your feedback!

**Comment by MostHated** [comment]
I had went back and edited my post at the top. I was able to get it working just fine. Here was my edit - It looks like all is working well for me with the change below.

Edit ------- It looks like, for the last part of this post, within the Token class, changing it to the following fixed it: There are still the few other things below, which are just small tweaks. I am not sure if having to do HiveStore.create() as opposed to just using HiveStore() is intended. It's getting a bit late for me tonight, but I can try to do a PR tomorrow unless you get to it first.
 
```dart
Token.fromMap(Map<dynamic, dynamic> map)

// and also
Map<dynamic, dynamic> toMap()
```
